### PR TITLE
[1LP][RFR] New bottleneck tests

### PR DESCRIPTION
**Pull request by otsuman**
{{ pytest: cfme/tests/optimize/ -k "test_bottlenecks_summary_time_zone or test_bottlenecks_summary_show_host_events or test_bottlenecks_summary_event_groups"}}

### CODE REVIEW

File: utils/appliance/__init__.py [hunk 1]
```diff
@@ -1062,6 +1062,11 @@ def restore_database(self, database_path="/tmp/evm_db.backup", log_callback=None
 
         """
         log_callback('Restoring database')
+        if current_version() > '5.8':
```
Comment (psav) [inline utils/appliance/__init__.py]: We should be checking this against the appliance itself using ```self``` right? ```self.version```

Comment (otsuman) [inline utils/appliance/__init__.py]: done

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/optimize/test_bottlenecks.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def db_restore(temp_appliance_extended_db):
     db_storage_ssh = SSHClient(hostname=db_storage_hostname, **conf.credentials['bottlenecks'])
     with db_storage_ssh as ssh_client:
         # Different files for different versions
-        ver = "_56" if current_version() < '5.7' else ""
+        ver = "_58" if current_version() > '5.7' else ""
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: REQUIRED: please get current_version from appliance

File: cfme/tests/optimize/test_bottlenecks.py [hunk 3]
```diff
@@ -107,4 +107,50 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
         # Changing time zone
         view.report.time_zone.fill('(GMT-04:00) La Paz')
         row = view.report.event_details[0]
-        assert row[0].text == (db_row[0][0] - timedelta(hours=4)).strftime("%m/%d/%y %H:%M:%S BOT")
+        assert row[0].text == (db_row[0][0] - timedelta(hours=4)).strftime("%m/%d/%y %H:%M:%S -04")
+
+
+@pytest.mark.tier(2)
+def test_bottlenecks_summary_event_groups(temp_appliance_extended_db, db_restore, db_tbl,
+                                          db_events):
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: REQUIRED: please add description to tests

File: cfme/tests/optimize/test_bottlenecks.py [hunk 4]
```diff
@@ -107,4 +107,50 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
...
+
+
+@pytest.mark.tier(2)
+def test_bottlenecks_summary_show_host_events(temp_appliance_extended_db, db_restore, db_events):
+    with temp_appliance_extended_db:
+        view = navigate_to(Bottlenecks, 'All')
+        view.summary.show_host_events.fill(False)
+        # Checking that events with value 'Host / Node' absent in table
+        events = view.summary.chart.get_events()
+        assert not sum(1 for event in events if event.type == 'Host')
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: :+1:

File: cfme/tests/optimize/test_bottlenecks.py [hunk 5]
```diff
@@ -107,4 +107,50 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
...
+        view.summary.show_host_events.fill(True)
+        events = view.summary.chart.get_events()
+        # Compare number of events in chart with number of rows in db
+        assert len(events) == db_events.count()
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: I'm not sure I understand where you generate events. I don't think it is good idea to rely on presence of some events in appliance db

File: cfme/tests/optimize/test_bottlenecks.py [hunk 6]
```diff
@@ -107,4 +107,50 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
...
+
+
+@pytest.mark.tier(2)
+def test_bottlenecks_summary_time_zome(temp_appliance_extended_db, db_restore, db_tbl, db_events):
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: REQUIRED: typo

File: cfme/tests/optimize/test_bottlenecks.py [hunk 7]
```diff
@@ -107,4 +107,50 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
...
+    with temp_appliance_extended_db:
+        view = navigate_to(Bottlenecks, 'All')
+        events = view.summary.chart.get_events()
+        # Selecting row by uniq value
+        db_row = db_events.filter(db_tbl.message == events[0].message)
+        # Compare event timestamp with db
+        assert events[0].time_stamp == db_row[0][0].strftime("%Y-%m-%d %H:%M:%S UTC")
```
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: it makes sense to use variable for 
event = events[0]
db_record_timestamp = db_row[0][0]

File: utils/appliance/__init__.py [hunk 8]
```diff
@@ -1062,6 +1062,12 @@ def restore_database(self, database_path="/tmp/evm_db.backup", log_callback=None
             log_callback(msg)
             raise ApplianceException(msg)
 
+        if self.version > '5.8':
+            status, output = self.ssh_client.run_command("fix_auth --databaseyml -i {}".format(
+                conf.credentials['database'].password), timeout=45)
+            if status != 0:
```
Comment (izapolsk) [inline utils/appliance/__init__.py]: I would use SSHResult.failed here

File: cfme/tests/optimize/test_bottlenecks.py (hunk 5 above)
Comment (otsuman) [inline cfme/tests/optimize/test_bottlenecks.py]: It takes  ~2 days to gather data. So the best way for checking these controls it to restore db

File: cfme/tests/optimize/test_bottlenecks.py (hunk 7 above)
Comment (otsuman) [inline cfme/tests/optimize/test_bottlenecks.py]: I think it is readable enough and not frequently used

File: utils/appliance/__init__.py (hunk 8 above)
Comment (otsuman) [inline utils/appliance/__init__.py]: I don't see a reason whyI should use it here

File: cfme/tests/optimize/test_bottlenecks.py (hunk 3 above)
Comment (otsuman) [inline cfme/tests/optimize/test_bottlenecks.py]: done

File: cfme/tests/optimize/test_bottlenecks.py (hunk 7 above)
Comment (izapolsk) [inline cfme/tests/optimize/test_bottlenecks.py]: I don't plan to insist. But consider using variable with meaningful name when you use complex constructions like db_row[0][0] more than 2 times.
here you use for example db_row[0][0] 3 times. 
btw, variable db_row[0][0] doesn't bring any meaning about what is in it. So, when you see it first time in code (assignment statement) it is more or less obvious but later it looses its meaning especially if there are several variables with similar names.

File: cfme/tests/optimize/test_bottlenecks.py (hunk 4 above)
Comment (mshriver) [inline cfme/tests/optimize/test_bottlenecks.py]: ^ditto

File: utils/appliance/__init__.py (hunk 8 above)
Comment (mshriver) [inline utils/appliance/__init__.py]: more pythonic, plain english instead of comparators and integers ;)

**Comment by mshriver** [comment]
**Required**
Fix two QuantifiedCode issues, should be real quick and easy ones (docstring and kwargs.get default)

File: cfme/tests/optimize/test_bottlenecks.py [hunk 9]
```diff
@@ -107,4 +109,53 @@ def test_bottlenecks_time_zome(temp_appliance_extended_db, db_restore, db_tbl, d
...
+@pytest.mark.tier(2)
+def test_bottlenecks_summary_time_zone(temp_appliance_extended_db, db_restore, db_tbl, db_events):
+    """ Checks time zone selectbox in summary tab. It should change time zone of events in chart """
+    with temp_appliance_extended_db:
+        view = navigate_to(Bottlenecks, 'All')
+        events = view.summary.chart.get_events()
+        # Selecting row by uniq value
+        db_row = db_events.filter(db_tbl.message == events[0].message)
+        # Compare event timestamp with db
+        assert events[0].time_stamp == db_row[0][0].strftime("%Y-%m-%d %H:%M:%S UTC")
```
Comment (mshriver) [inline cfme/tests/optimize/test_bottlenecks.py]: This looks like ISO with UTC: 
https://github.com/ManageIQ/integration_tests/blob/master/utils/timeutil.py#L19

import parsetime and use that instead.  Doesn't really help for the test above unfortunately, where you're working with a different offset not in this util.

File: widgetastic_manageiq.py [hunk 10]
```diff
@@ -1589,8 +1589,9 @@ def __repr__(self):
             params = ", ".join(["{}={}".format(attr, getattr(self, attr)) for attr in attrs])
             return "TimelinesEvent({})".format(params)
 
-    def __init__(self, parent, logger=None):
+    def __init__(self, parent, locator=None, logger=None):
         super(TimelinesChart, self).__init__(parent=parent, logger=logger)
+        self.locator = locator or '//div[contains(@class, "timeline-container")]'
```
Comment (mshriver) [inline widgetastic_manageiq.py]: **Required**

It looks like the whole reason you've made this change is to support adding the chart with a `timeline-pf-chart` div.
(https://github.com/ManageIQ/integration_tests/pull/4627/files#diff-e25760b09cb1951870a11138d80a45f3R34)


It seems this chart element is patternfly, which means this change isn't needed for manageiq:
https://github.com/patternfly/patternfly-timeline/blob/27f0a72733238ebebf8ab69a5803be2ad7e6441c/src/timeline.js#L57

Unfortunately this is a major change if I'm right.

@mfalesni @izapolsk Thoughts? This should be added to patternfly and we need to bump that version before accepting this PR.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: widgetastic_manageiq.py (hunk 10 above)
Comment (mfalesni) [inline widgetastic_manageiq.py]: @mshriver @otsuman to save time, I would personally keep it here for now and then backport it to PF, then remove and bump ...

**Pull request merged**